lib/note-list/index.tsx

```diff
@@ -496,14 +491,12 @@ const { recordEvent } = tracks;
 
 const mapStateToProps = ({
   appState: state,
-  ui: { filteredNotes },
+  ui: { filteredNotes, note },
   settings: { noteDisplay },
 }) => {
   const tagResultsFound = getMatchingTags(state.tags, state.filter).length;
-
-  const noteIndex = Math.max(state.previousIndex, 0);
-  const selectedNote = state.note ? state.note : filteredNotes[noteIndex];
-  const selectedNoteId = get(selectedNote, 'id', state.selectedNoteId);
+  const selectedNote = note;
+  const selectedNoteId = get(selectedNote, 'id', null);
```

Comment:
we should be free to use optional-chaining, but I haven't confirmed

```ts
const selectedNoteId = selectedNote?.id;
```

if it matters that it be `null` in the absence of a `selectedNote` then we only need add that default

```ts
const selectedNoteId = selectedNote?.id || null;
```

we could use the nullish coalescing operator to prevent collapsing a falsey value `…?.id ?? null` but since we know that `T.EntityId` is a string and an empty string is an invalid entity id we can leave the `||`

Reply:
Ohhhhhhhh wow.  Optional chaining is my fav new feature no more `obj && obj.prop`. I so happy this is a thing!!

Reply:
Stacking: `let customerCity = customer.details?.address?.city;`!!!!

Reply:
just know that it's not the same as `a && a.b && a.b.c` because it's only checking for `null` and `undefined` and not any falsey value. this is stronger in terms of safety but weaker in terms of what we usually do